server/src/internalClusterTest/java/org/opensearch/action/bulk/BulkIntegrationIT.java

```diff
@@ -218,4 +219,32 @@ public void testDeleteIndexWhileIndexing() throws Exception {
         assertFalse(thread.isAlive());
     }
 
+    public void testDocIdTooLong() {
+        String index = "testing";
+        createIndex(index);
+        String invalidId = String.join("", Collections.nCopies(513, "a"));
+
+        // Index Request
+        IndexRequest indexRequest = new IndexRequest(index).id(invalidId);
+        indexRequest.source(Collections.singletonMap("foo", "baz"));
+        validateDocIdLimit(() -> client().prepareBulk().add(indexRequest).get());
+
+        // Update Request
+        UpdateRequest updateRequest = new UpdateRequest(index, invalidId).doc("reason", "no source");
+        validateDocIdLimit(() -> client().prepareBulk().add(updateRequest).get());
+    }
+
+    private void validateDocIdLimit(Runnable runner) {
```

Comment:
`expectThrows` should also do what you are trying to achieve here.

```java
            ActionRequestValidationException ex = expectThrows(ActionRequestValidationException.class, () -> client().prepareBulk().add(indexRequest).get());
```

Reply:
It sort of does but the request validation can happen due to multiple reasons. This method validates that the error message for docID length is also present.

Reply:
This is a nit, but you have a reference to the ActionRequestValidationException so should be able to?
```java
            ActionRequestValidationException ex = expectThrows(ActionRequestValidationException.class, () -> client().prepareBulk().add(indexRequest).get());
            assertEquals(
                1,
                ex.validationErrors().stream().filter(msg -> msg.contains("is too long, must be no longer than 512 bytes but was")).count()
            );
```